src/prism.c

```diff
@@ -12825,6 +12833,14 @@ parse_variable_call(pm_parser_t *parser) {
                 return (pm_node_t *) pm_local_variable_read_node_create(parser, &parser->previous, 0);
             }
         }
+        else if (!parser->current_scope->closed && pm_token_is_it(parser->previous.start, parser->previous.end)) {
```

Comment:
```suggestion
        } else if ((parser->version != PM_OPTIONS_VERSION_CRUBY_3_3_0) && !parser->current_scope->closed && pm_token_is_it(parser->previous.start, parser->previous.end)) {
```

We need to gate this with 3.3.0.